gravitee-apim-console-webui/src/management/api/portal/subscriptions/components/dialogs/creation/api-portal-subscription-creation-dialog.component.spec.ts

```diff
@@ -207,7 +214,199 @@ describe('Subscription creation dialog', () => {
         expect(await harness.isCustomApiKeyInputDisplayed()).toBeFalsy();
       });
     });
-    describe('With custom apiKey disabled', () => {
+    describe('With custom API Key enabled and shared apiKey enabled', () => {
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          declarations: [TestComponent],
+          imports: [ApiPortalSubscriptionsModule, NoopAnimationsModule, GioHttpTestingModule, MatIconTestingModule],
+          providers: [
+            { provide: SubscriptionService },
+            {
+              provide: InteractivityChecker,
+              useValue: {
+                isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+                isTabbable: () => true, // This traps tabbable checks and so avoid warnings when dealing with
+              },
+            },
+            {
+              provide: 'Constants',
+              useFactory: () => {
+                const constants = CONSTANTS_TESTING;
+                set(constants, 'env.settings.plan.security', {
+                  customApiKey: { enabled: true },
+                  sharedApiKey: { enabled: true },
+                });
+                return constants;
+              },
+            },
+          ],
+        });
+        fixture = TestBed.createComponent(TestComponent);
+        httpTestingController = TestBed.inject(HttpTestingController);
+        fixture.detectChanges();
+        component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+      });
+
+      afterEach(() => {
+        jest.clearAllMocks();
+        httpTestingController.verify();
+      });
+
+      it('should be able to select API Key mode EXCLUSIVE and custom API Key', async () => {
+        const applicationWithClientId = fakeApplication({
+          id: 'my-app',
+          name: 'withClientId',
+          settings: { app: { client_id: 'clientId' } },
+          api_key_mode: ApiKeyMode.UNSPECIFIED,
+        });
+        const planV4 = fakePlanV4({ apiId: 'my-api', mode: 'STANDARD', security: { type: 'API_KEY' }, generalConditions: undefined });
+        component.plans = [planV4];
+        component.availableSubscriptionEntrypoints = [];
+
+        jest.spyOn(TestBed.inject(SubscriptionService), 'getApplicationSubscriptions').mockReturnValue(
```

Comment:
why mock this service and not use httpTestingController ?

Reply:
You're right

https://media3.giphy.com/media/1JTy9isIm0u0OqePP1/200w.gif?cid=5a38a5a2ty8cpsw0vdvv9ukgbbduffz0edujzm12cb0w4saq&ep=v1_gifs_search&rid=200w.gif&ct=g